cli/src/semgrep/engine.py

```diff
@@ -26,11 +32,20 @@ def decide_engine_type(
         requested_engine: Optional["EngineType"] = None,
         scan_handler: Optional[ScanHandler] = None,
         git_meta: Optional[GitMeta] = None,
+        run_secrets: bool = False,
     ) -> "EngineType":
         """Select which Semgrep engine type to use if none is explicitly requested.
 
         Considers settings from Semgrep Cloud Platform and version control state.
         """
+        # Change default to pro-engine intrafile if secrets was requested.
+        # Secrets is built into pro-engine, but any pro-setting should work.
+        if requested_engine is None and run_secrets:
+            requested_engine = cls.PRO_LANG
+        elif run_secrets and requested_engine is cls.OSS:
+            # Should be impossible if the CLI gates impossible arguemnet combinations.
```

Comment:
all those extra checks everywhere are annoying ...
It's like 5 times we check whether you got secrets and OSS ...